sdk/go/pulumi/types.go

```diff
@@ -728,6 +746,11 @@ func awaitInputs(ctx context.Context, v, resolved reflect.Value) (bool, bool, []
 
 	contract.Assertf(valueType.AssignableTo(resolved.Type()), "%s not assignable to %s", valueType.String(), resolved.Type().String())
 
+	if v.CanInterface() && v.Type().Implements(resourceType) {
+		resolved.Set(v)
+		return true, false, nil, nil
```

Comment:
Shouldn't deps here include v.urn?

Reply:
No--including the URNs of resource instances encountered is inconsistent with the other SDKs.